### DISCUSSION TITLE
How to use metadata field in templates like `path`?

### DISCUSSION THREAD

**Discussion by 0akarma** [vectordotdev/vector#15961] [Q&A] [answered]
I have noticed that vector have supported [setting metadata field like](https://vector.dev/highlights/2022-10-04-0-25-0-upgrade-guide/#metadata-function-deprecation) `%kafka.partition = del(.partition)`, but how can I use this metadata field in the template syntax?
I want to split file by kafka partition like `path = "/output/%Y-%m-%d-%H-{{ %kafka.partition) }}.json.gz"`

**Comment by jszwedko** [comment]
Hi @0akarma !

I believe that should work. I take it that it is not working for you though?

**Reply by spencergilbert** [reply]
`%kafka.partition` implies it's stored in the metadata, where the `partition_key` is stored on the event itself (at least until log namespacing is enabled).

**Reply by jszwedko** [reply]
Ah, I see. I may have misread their question. I thought they were setting `%kafka.partition` in a `remap` transform (via `%kafka.partition = del(.partition)`) but maybe they were just offering that as an example.

**Reply by spencergilbert** [reply]
Oh, I also might have misread it then 😅 - then I would expect it to work as provided as well (removing the stray `)` that got added after the word partition though)

**Comment by spencergilbert** [comment, marked as answer]
Hey @0akarma - I believe the configuration you want to use is:

```toml
path = "/output/%Y-%m-%d-%H-{{ .partition }}.json.gz"
```
